### PR TITLE
170378412 Implement error handling for route handlers

### PR DESCRIPTION
**Pull request by kossy360**
- sets up a global error handler
- integrates an extension of the node Error object to handle application requirements
- adds an async wrapper for gracefully handling exceptions
- adds user-defined constraints to tables
- modifies .gitignore
- adds migration script for dropping all local databases

[Finishes 170378412]

### CODE REVIEW

**Review by Fiyiin — COMMENTED** [review]
is declaring a new Error object different from Node's Error object or an override?

File: src/processor/organization/auth.js [hunk 1]
```diff
@@ -1,55 +1,52 @@
 import crypto from 'crypto';
 import bcrypt from 'bcrypt';
 import { getDB } from '../../db/local-models';
+import { Organization } from '../../db/models';
+import errorHandler from '../../error-handler';
+
+const Error = errorHandler.withDetails;
+
+const checkIfOrgExists = async (id) => {
+  const organization = await Organization.findOne({ where: { id } });
```
Comment (eben-k) [inline src/processor/organization/auth.js:10]: We are breaking the one db call rule here, I guess.

File: src/scripts/migration.js [hunk 2]
```diff
@@ -42,6 +42,15 @@ const migrateAllLocal = async () => {
...
 
+const dropAllLocal = async () => {
+  const organizations = await model.Organization.findAll({});
+
+  for (const organization of organizations) {
+    // eslint-disable-next-line no-await-in-loop
+    await model.sequelize.query(`DROP DATABASE ${organization.id}`);
+  }
+};
+
```
Comment (eben-k) [inline src/scripts/migration.js:53]: Good work here creating an automatic db drop script

File: src/utils/async-wrapper.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import Error from '../error-handler';
+
+export default (func) => (req, res, next) => func(req, res, next)
+  .catch((err) => (err.isCustomError ? next(err) : next(new Error(err))));
```
Comment (eben-k) [inline src/utils/async-wrapper.js:4]: Funny that the great  ```Async``` is only 4 lines.

**Review by eben-k — APPROVED** [review]
Great work all over.... would wait for the walk-through although I pretty much get what has been done

**Comment by kossy360** [comment]
> is declaring a new Error object different from Node's Error object or an override?

@Fiyiin The new error object is an extension of Node's error object

**Review by Fiyiin — APPROVED** [review]
LGTM

**Pull request merged**